warpscript/utapi/createCheckpoint.mc2

```diff
@@ -0,0 +1,155 @@
+{
```

Comment:
This macro looks much like the snapshot one, is there a way to reuse parts of the same code somehow? I'm a noob in mc2 so just take this as a suggestion.

Reply:
There's definitely some generalization that can be done, but as I'm still a noob as well, I've tried to write it in a straightforward manner to try and salvage what readability I can. When I add the repair process in a later PR, I plan to do some refactoring as the requirements will be a bit more clear.